drv/fpga-api/src/lib.rs

```diff
@@ -257,14 +257,19 @@ impl FpgaUserDesign {
         T: AsBytes + FromBytes,
     {
         let mut v = T::new_zeroed();
-        self.server.user_design_read(
-            self.device_index,
-            addr.into(),
-            v.as_bytes_mut(),
-        )?;
+        self.read_bytes(addr.into(), v.as_bytes_mut())?;
```

Comment:
Nit: you don't need `addr.into()` here, since `read_bytes` will accept it as-is.

Reply:
Good point, I had not noticed that. Cleaned up in latest commit.